### ISSUE TITLE
Support record instance matching with `annotation(choicesAllMatching = true)`

### ISSUE THREAD

**Issue by Maxwellfire** [OpenModelica/OpenModelica#11246]
### Description

The [modelica spec](https://specification.modelica.org/master/inheritance-modification-and-redeclaration.html#annotation-choices-for-suggested-redeclarations-and-modifications) says that GUIs should support matching record instances in a configuration dropdown. 

> For a record variable the included elements shall include matching record constants and calls of matching record constructors (matching classes as for replaceable elements)

Currently, this is not supported by OMEdit.

### Steps to Reproduce

The spec gives a clear example of what should be supported:

>[Example: Demonstrating the choicesAllMatching = true annotation for parameter records.
>```
>record Medium
>  parameter SI.Density rho "Density";
>  …
>end Medium;
>record Air_30degC = Medium(rho = 1.149, …);
>constant Medium MyAir = Medium(rho = 1.1, …);
>model OpenTank
>  parameter Medium medium = Medium() annotation(choicesAllMatching = true);
>end OpenTank;
>```
>
>The choices for medium shall include Medium(), Air_30degC(), and MyAir. If Medium() is chosen it is necessary to also set its rho-parameter.]

### Version and OS

- OpenModelica 1.23.0~dev-2-ga714efa
- OMSimulator 2.1.1~236-g634b019-linux-notlm-debug
- Kubuntu 23.04

**Comment by casella** [comment]
@Maxwellfire you are right. Extending what we already have for replaceable classes should be easy. @perost can you comment on that?

**Comment by perost** [comment]
> @Maxwellfire you are right. Extending what we already have for replaceable classes should be easy. @perost can you comment on that?

`getAllSubtypeOf` already works for finding compatible record classes I think, but for record constants we need to do something different. In `getAllSubtypeOf` we take shortcuts that significantly improve the performance since we know we only need to look up base classes, so naively resolving the types of all package constants might be too slow.

So supporting record constants might require a bit of work, but supporting record classes might just be a matter of using `getAllSubtypeOf` in OMEdit similar to how replaceable classes are handled.

**Comment by casella** [comment]
@perost I guess we should fix this for 1.26.0
